examples/features/effect-handlers.effekt.md

```diff
@@ -0,0 +1,87 @@
+# Effects and handlers
+
+For getting started with effects and handlers, we define the popular exception effect, consisting just of one effect operation `throw`:
+
+```
+interface Exception {
+  def throw(msg: String): Nothing
+}
+```
+
+Each effect declaration starts with the `interface` keyword, followed by the name of the effect. Enclosed in curly braces, effect operations belonging to this effect are declared with their signature.
+
+For envoking an effect, the respective effect operations can be called with the usual function call syntax, prepended by `do`:
+
+```
+def div(a: Double, b: Double) = 
+  if (b == 0.0) { do throw("division by zero") }
+  else { a / b }
+```
+
+The return type of non-recursive functions can be left out and is inferred. In this it is `Double / { Exception }`, that is, the return type `Double` followed by a slash and the effects that need to be handled by the calling context.
+
+Whereas operations introduce effects, handlers offer a way of discharging them:
+
+```
+def unsafeDiv(a: Double, b: Double): Double / {} =
+  try {
+    div(a, b)
+  } with Exception {
+    def throw(msg) = {
```

Comment:
Good question!

If we had defined the `Exception` effect as follows

```
effect Exception(msg: String): Nothing
```

we could have used the syntax you described. However, this is IMO in a bit of a weird place right now because of the different  keyword (`effect` vs `interface`). I am therefore reluctant to mention it. I'll give it some thought.

Reply:
Oh interesting, okay! That clears up a lot of confusion I had. I thought `effect Exception(msg: String): Nothing` was a direct alias to `interface Exception { def Exception(msg: String): Nothing }` and couldn't figure out why it wasn't working...